### PR TITLE
refactor: narrow type definition of branch config

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes
- Introduce 2 new interfaces
1. `NarrowedBranchConfig` -> config fields that are used when processing a branch
2. `UpdateCacheConfig` -> config fields that are necessary to verify if a branch update is necessary

## Context
- Update cache fingerprint similar to the extract cache fingerprinting 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by RahulGautamSingh** [comment]
WIP: 
(todos)
1. Create an issue to outline the flow of  update cache process
1. Refactor `processBranch` logic in a separate PR
2. Refactor `ensureBranch` and `ensurePR`

File: lib/workers/types.ts [hunk 1]
```diff
@@ -127,6 +130,173 @@ export interface BranchConfig
...
+  warnings?: ValidationMessage[];
+  upgrades: BranchUpgradeConfig[];
+  packageFiles?: Record<string, PackageFile[]>;
+  npmLock?: string;
+  yarnLock?: string;
+  skipInstalls?: boolean;
+  constraints?: any;
+  ignoreScripts?: boolean;
+  postUpdateOptions?: string[];
+  managerData?: NpmManagerData;
```
Comment (viceice) [inline lib/workers/types.ts:151]: this is wrong, use `Record` instead. this is different for each manager

File: lib/workers/types.ts [hunk 2]
```diff
@@ -127,6 +130,173 @@ export interface BranchConfig
   isConflicted?: boolean;
 }
 
+export interface NarrowedBranchConfig extends Record<string, any> {
```
Comment (viceice) [inline lib/workers/types.ts:133]: @zharinov looks the wrong way to me.

 i think duplicate all those properties makes further null checks more complicated. also maintainability will be more worse.

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/workers/types.ts (hunk 2 above)
Comment (RahulGautamSingh) [reply lib/workers/types.ts:133]: We don't need to implement narrowing. I created the narrowed config only to make it easy to track fields that are necessary for caching.

**Pull request closed without merging**